src/hwc/compile/luna/luna.go

```diff
@@ -0,0 +1,238 @@
+package luna
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cloudfoundry/libbuildpack"
+)
+
+type Luna struct {
+	Log                    *libbuildpack.Logger
+	BuildDir               string
+	LunaDir                string
+	credentials            *LunaCredentials
+	ClientPrivateKeyPath   string
+	ClientCertificatePath  string
+	ServerCertificatesPath string
+}
+
+type Client struct {
+	Certificate string
+	PrivateKey  string `json:"private-key"`
+}
+
+type Server struct {
+	Name        string
+	Certificate string
+}
+
+type Group struct {
+	Label   string
+	Members []string
+}
+
+type LunaCredentials struct {
+	Client  Client
+	Servers []Server
+	Groups  []Group
+}
+
+func NewLuna(log *libbuildpack.Logger, buildDir string) *Luna {
+	l := new(Luna)
+	l.Log = log
+	l.BuildDir = buildDir
+	l.LunaDir = filepath.Join(l.BuildDir, ".cloudfoundry", ".luna")
+	l.ClientPrivateKeyPath = filepath.Join(l.LunaDir, "clientPrivateKey.pem")
+	l.ClientCertificatePath = filepath.Join(l.LunaDir, "clientCertificate.pem")
+	l.ServerCertificatesPath = filepath.Join(l.LunaDir, "serverCertificates.pem")
+	return l
+}
+
+func validateCredentials(credentials *LunaCredentials) bool {
+	if credentials.Client.Certificate == "" || credentials.Client.PrivateKey == "" {
+		return false
+	}
+	if len(credentials.Servers) == 0 {
+		return false
+	}
+	for _, server := range credentials.Servers {
+		if server.Name == "" || server.Certificate == "" {
+			return false
+		}
+	}
+	for _, group := range credentials.Groups {
+		if group.Label == "" || len(group.Members) == 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func (l *Luna) parseVCAPServices() {
```

Comment:
This function is only looking at the structure of the `VCAP_SERVICES` entry, and that structure seems fairly generic. Because the config files for Luna should only be added if there is actually a Luna service bound to this app, it is important that `parseVCAPServices` does not have false-positives.

We'd prefer if you looked for a particular name or part of a name. For example:
In the `go-buildpack`, appdynamics looks for a service named `appdynamics`: https://github.com/cloudfoundry/go-buildpack/blob/master/src/go/hooks/appdynamics.go#L79
In the `nodejs-buildpack`, snyk looks for a service whose name contains `snyk`: https://github.com/cloudfoundry/nodejs-buildpack/blob/master/src/nodejs/hooks/snyk.go#L237

Reply:
I will change this to check if the name contains ```luna``` or ```hsm```.